src/VisualStudio/CSharp/Impl/Options/Formatting/CodeStylePage.cs

```diff
@@ -11,7 +19,502 @@ internal class CodeStylePage : AbstractOptionPage
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.PreferConditionalExpressionOverAssignment));
+            editorconfig.Append("dotnet_style_prefer_conditional_expression_over_return = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.PreferConditionalExpressionOverReturn));
+
+            editorconfig.AppendLine();
+            // C# Coding Conventions
+            editorconfig.AppendLine("###############################");
+            editorconfig.AppendLine("# C# Coding Conventions       #");
+            editorconfig.AppendLine("###############################");
+
+            // Var preferences
+            editorconfig.AppendLine("# var preferences");
+            editorconfig.Append("csharp_style_var_for_built_in_types = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.UseImplicitTypeForIntrinsicTypes));
+            editorconfig.Append("csharp_style_var_when_type_is_apparent = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.UseImplicitTypeWhereApparent));
+            editorconfig.Append("csharp_style_var_elsewhere = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.UseImplicitTypeWherePossible));
+
+            editorconfig.AppendLine();
+            // Expression-bodied members
+            editorconfig.AppendLine("# Expression-bodied members");
+            editorconfig.Append("csharp_style_expression_bodied_methods = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedMethods));
+            editorconfig.Append("csharp_style_expression_bodied_constructors = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedConstructors));
+            editorconfig.Append("csharp_style_expression_bodied_operators = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedOperators));
+            editorconfig.Append("csharp_style_expression_bodied_properties = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedProperties));
+            editorconfig.Append("csharp_style_expression_bodied_indexers = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedIndexers));
+            editorconfig.Append("csharp_style_expression_bodied_accessors = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedAccessors));
```

Comment:
there seems to be a lot of redundancy here.  especially around you having to say ```Append("csharp_style_expression_bodied_accessors = ")```.  We already know the editor config name.  It's specificied in CSharpCodeStyleOptions.PreferExpressionBodiedAccessors.StorageLocations.  

I think you could simplify all of this by just listing the set of options, and having code that introspects the option and prints out the editorconfig line for it.